### PR TITLE
DAOS-9097 test: Allow the build to continue on valgrind error.

### PR DESCRIPTION
**Pull request by ashleypittman**
Change the behaviour of NLT not to exit non-zero if there are
valgrind issues reported.  These errors are flagged by Jenkins
when the build completes stopping the build is not required,
rater have NLT add them as a test failure, then it's up to the
build configuration if jobs stop or not.

Also detect the case where exceptions mean that an attempt
is made to stop the server whilst dfuse instances are running,
if this happens then stop the dfuse instance before stopping
the server, and again register this as a test failure.

Fix a logparse bug where a recent change meant that client
logs over 100Mb in size could not be parsed,

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
Please address merge conflict.

File: src/tests/ftest/cart/util/cart_logparse.py [hunk 1]
```diff
@@ -601,8 +596,7 @@ def _load_pids(self):
         for line in self._fd:
             fields = line.split(None, 8)
             index += 1
-            l_pid = None
-            if len(fields) < 6 or len(fields[0]) != 17:
+            if len(fields) < 6 or len(fields[0]) != 17 or fields[0][2] != '/':
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_logparse.py:599]: What does '/' correspond to here? Or rather, is there an example output for this file/line?

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/cart/util/cart_logparse.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/cart/util/cart_logparse.py:599]: This is just bringing it line with L575 above, example output would be any DAOS log file.

**Review by kccain — APPROVED** [review]

**Comment by sylviachanoiyee** [comment]
@ashleypittman , build #15 has a lot of tests failing with
unexpected EOF while parsing
Can you take a look?

**Comment by ashleypittman** [comment]
> @ashleypittman , build #15 has a lot of tests failing with unexpected EOF while parsing Can you take a look?

That was an issue with the skip-list file wasn't it?  All PRs were affected for a while.

**Comment by sylviachanoiyee** [comment]
> > @ashleypittman , build #15 has a lot of tests failing with unexpected EOF while parsing Can you take a look?
> 
> That was an issue with the skip-list file wasn't it? All PRs were affected for a while.

Yes, and it's been fixed, but looks like build #16 failed coverity which stopped hw testing from running. Coverity issue is worked around and should be skipped until John M comes back from vacation. Maybe retrigger?

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**